src/constructLayoutEngine.js

```diff
@@ -63,56 +72,97 @@ export function constructLayoutEngine({
   function arrangeDomElements() {
     const path = location[resolvedRoutes.mode === "hash" ? "hash" : "pathname"];
 
-    // TODO memoize this for perf
-    const matchedRoutes = matchRoute(resolvedRoutes, path);
```

Comment:
Removed this for a couple reasons:

1. Perf - doing both matchRoutes and recurseRoutes is double iteration
2. Removing dom elements - we need to recurse all routes (not just matched ones) so that we know which dom elements are no longer active.